### ISSUE TITLE
[FEATURE] Make it easy to build the project in different Docker environments with a bash script

### ISSUE THREAD

**Issue by okram** [opensearch-project/security-analytics#19]
**Is your feature request related to a problem?**

`CONTRIBUTING.md` explains how to use Docker to build the project. Reading and understanding the explanation takes time and effort. A configurable shell script that encapsulates the idea would make the developer's experience smoother. (e.g. "What's Docker? What's all this about mounting volumes? Why can't you just 'make it happen'?").

**What solution would you like?**

A configurable shell script. Please see work in progress appended below.

**What alternatives have you considered?**

It would be best if `opensearch-darwinx64-2.1.0-SNAPSHOT.zip` existed. Barring that, the recommended feature is quite nice. Moreover, its more general in that it provides a glimpse of how the project can  use Docker down the road in interesting (currently unrealized) ways.

**Do you have any additional context?**

The shell script I've built locally that is currently a work in progress.

```bash
#!/bin/bash

# Usage: gradlex build
#
# 1.) Mounts current directory into specified DOCKER_IMAGE (default: gradle:7.4.2-jdk11-focal)
# 2.) Mounts GRADLE_CACHES (default: ~/docker-gradle-caches) into Docker image's .gradle/caches.
# 3.) Evaluates gradle $1 (where any gradle task can be specified in as an argument to the script).


if [ -z "${GRADLE_CACHES}" ]; then
  GRADLE_CACHES=~/docker-gradle-caches
fi

if [ -z "${DOCKER_IMAGE}" ]; then
  docker pull gradle:7.4.2-jdk11-focal
  DOCKER_IMAGE=gradle:7.4.2-jdk11-focal
fi

echo "Building in Docker image ${DOCKER_IMAGE}"
echo " (Gradle build caches: ${GRADLE_CACHES})"

docker run -t -i -v $GRADLE_CACHES:/home/gradle/.gradle/caches --user gradle -v $(pwd):/docker/tmp -w /docker/tmp $DOCKER_IMAGE gradle $1
```

**Comment by getsaurabh02** [comment]
Thanks for sharing the context and details @okram.

Can we add these in form of guided steps to [Contributing.md](https://github.com/opensearch-project/security-analytics/blob/main/CONTRIBUTING.md) for this package. 

Also, once your script is ready, and if you think it will be useful for other other developers as well, we should add the generic script in build-tools, and share details in the  Contributing.md

**Comment by okram** [comment]
The example provided in the feature request is just a "this is the direction I'm thinking of taking it." I've added a more complete script in the subsequent pull request. I didn't update `CONTRIBUTING.md` as there is another PR with `CONTRIBUTING.md` updates. If you plan to merge the #18, then perhaps easiest if you could then just add the following text before pushing. If you don't want to merge #18, then I can update `CONTRIBUTING.md` with a reissue of this PR.

--

A self-contained build script intended for MacOSX developers is available at `bin/gradlex`.

```bash
$ bin/gradlex --help
GradleX: Gradle Built in the Image of Docker
  DOCKER_IMAGE (default: gradle:7.4.2-jdk11-focal)
  GRADLE_CACHES (default: /Users/marko/docker-gradle-caches)
Arguments:
  -h, --help: this help message
  <task>: gradle /build.gradle <task>
 ```
